lib/form/paragraph3.flow

```diff
@@ -143,24 +153,36 @@ renderParagraph(
 	words : [ParaWord] = map(list2array(glued), makeParaWord);
 
 	aw = make(0.0);
+	width = if (widthManager == dummyWigiParagraphWidthManager) make(LARGE) else widthManager.width;
+	// height = make(0.0);
 	update = make(0);
 	linesB : DynamicBehaviour<[ParaLine]> = make([]);
 
-	renderParaLinesFn = \lines, storyWidth -> renderParaLines(
-		storyWidth, lines, tightWidth, alignment, interlineSpacing,
-		topLineBaseline, interactiveStyles, rtl, isParagraphMarked
-	);
+	optFn = \lines -> {
+		lasti = length(lines) - 1;
+		// None() for Justify, last line alignment overridden.
+		optzds = mapi(lines, \i, line -> getOptimizedLine(line, i != lasti && alignment==None()));
```

Comment:
optzds is used only once, why we need it at all?
anyway, it explain nothing because the name is not self-explaining

Reply:
I'm afraid it is copy-paste from renderParaLines, we should extract the common code into separate fn and use in both places